src/hera/workflows/script.py

```diff
@@ -901,16 +873,18 @@ def transform_script_template_post_build(
         self, instance: "Script", script: _ModelScriptTemplate
     ) -> _ModelScriptTemplate:
         """A hook to transform the generated script template."""
-        if _flag_enabled(_SCRIPT_ANNOTATIONS_FLAG):
-            if not script.env:
-                script.env = []
-            script.env.append(EnvVar(name="hera__script_annotations", value=""))
-            if self.outputs_directory:
-                script.env.append(EnvVar(name="hera__outputs_directory", value=self.outputs_directory))
-            if self.pydantic_mode:
-                script.env.append(EnvVar(name="hera__pydantic_mode", value=str(self.pydantic_mode)))
-            if _flag_enabled(_SCRIPT_PYDANTIC_IO_FLAG):
-                script.env.append(EnvVar(name="hera__script_pydantic_io", value=""))
+        script_env = []
+
+        if self.outputs_directory:
+            script_env.append(EnvVar(name="hera__outputs_directory", value=self.outputs_directory))
+        if self.pydantic_mode:
+            script_env.append(EnvVar(name="hera__pydantic_mode", value=str(self.pydantic_mode)))
+        if _flag_enabled(_SCRIPT_PYDANTIC_IO_FLAG):
+            script_env.append(EnvVar(name="hera__script_pydantic_io", value=""))
+
+        if script_env:
+            script.env = script_env
```

Comment:
Isn't this a behaviour change? We used to append to `script.env` if it existed, but now we overwrite it.

Reply:
No, it's equivalent, in the interest of avoiding an extra `if not script.env` within each `if` scope - we were always adding `hera__script_annotations` before so we could always initialise `script.env` if it was `None`/empty.

Reply:
In the case of user script environment variables, yes. Will fix

Reply:
Thank you for spotting! https://github.com/argoproj-labs/hera/pull/1289